### PR TITLE
Update subjects model

### PR DESCRIPTION
**Pull request by Nyakaru**
- update subjects model to accomodate more fields
- add department model

Screenshot from 2021-06-14 16-03-14

### CODE REVIEW

File: graphql/schema.graphql [hunk 1]
```diff
@@ -484,3 +529,8 @@ enum Gender {
     FEMALE
     NA
 }
+
```
Comment (MbugwaSami) [inline graphql/schema.graphql:532]: Status is Compusarly or Optional.

Comment (Nyakaru) [reply graphql/schema.graphql:532]: Everything I added is optional

Comment (MbugwaSami) [reply graphql/schema.graphql:532]: I mean the value should be Compusarly Subject or Optional Subject.

Comment (MbugwaSami) [reply graphql/schema.graphql:532]: What do Pending and Done represent? Don't think it's significant?

Comment (MbugwaSami) [reply graphql/schema.graphql:532]: @Nyakaru can you resolve this. we need to close this asap!

Comment (Nyakaru) [reply graphql/schema.graphql:532]: Sure on it

File: src/resolvers/mutations/users.js [hunk 2]
```diff
@@ -74,6 +74,17 @@ const signup = async (_, { input }, context, _info) => {
             return utils.sendErrorResponse(field, message);
         }
 
+        if (input['department']) {
+            message = 'Department does not exist';
```
Comment (MbugwaSami) [inline src/resolvers/mutations/users.js:78]: Are we changing how we adding users?

File: src/resolvers/mutations/subject.js [hunk 3]
```diff
@@ -75,6 +75,17 @@ const addSubject = async (_, { input }, context, _info) => {
...
+            field = 'department';
+            const departments = await context.prisma.school({ id: school }).department();
+
+            const depExists = departments.filter(item => item?.id == input['department']['connect']['id']);
+            if (!depExists.length) {
+                return utils.sendErrorResponse(field, message);
+            }
+        }
+
         message = 'Subject already exists in this school';
```
Comment (MbugwaSami) [inline src/resolvers/mutations/subject.js]: Can you add this validation for subjectAbv and subject code too.

File: src/resolvers/mutations/users.js (hunk 2 above)
Comment (Nyakaru) [reply src/resolvers/mutations/users.js:78]: No in the client it will remain as before

File: src/resolvers/mutations/subject.js (hunk 3 above)
Comment (Nyakaru) [reply src/resolvers/mutations/subject.js]: Sure

**Pull request merged**